### PR TITLE
Improve temperature reading

### PR DESCRIPTION
**Pull request by adam-3bian**
Show positive and negative temperatures in millidegrees Celsius according to page 28 of the AS621x Datasheet (DS000447v4).

### CODE REVIEW

**Review by marnovandermaas — COMMENTED** [review]
Thanks for contributing this here! I haven't been able to test this yet, but was wondering what the white space changes are for?

**Comment by adam-3bian** [comment]
> Thanks for contributing this here! I haven't been able to test this yet, but was wondering what the white space changes are for?

Agreed, the indentation around Debug::log should not have changed. I've swapped my configuration to use clang-format.

File: examples/all/i2c_example.cc [hunk 1]
```diff
@@ -22,7 +22,8 @@ static void read_temperature_sensor_value(Mmio<OpenTitanI2c> i2c,
 	i2c->blocking_write(0x48, buf, 1, false);
 	if (i2c->blocking_read(0x48, buf, 2u))
 	{
-		uint16_t temp = (buf[0] << 8) | buf[1];
+		int16_t regValue = (buf[0] << 8) | buf[1];
+		int64_t temp     = regValue * 1000000LL / 128;
 		Debug::log("The {} readout is {}", regName, temp);
```
Comment (HU90m) [inline examples/all/i2c_example.cc:27]: Units would be great to have in the log.

```suggestion
		Debug::log("The {} readout is {} microdegrees Celcius", regName, temp);
```

*Also, am I right in thinking the temperature is microdegrees not millidegrees after the conversion?*

**Review by HU90m — APPROVED** [review]
This is great thank you!

*For future me: the datasheet can be found [here](https://docs.rs-online.com/9186/A700000007240992.pdf).*

**Pull request merged**